astropy/io/fits/convenience.py

```diff
@@ -614,29 +618,32 @@ def append(filename, data, header=None, checksum=False, verify=True, **kwargs):
         # output to the empty object.
         #
         writeto(filename, data, header, checksum=checksum, **kwargs)
+
     else:
         hdu = _makehdu(data, header)
 
         if isinstance(hdu, PrimaryHDU):
             hdu = ImageHDU(data, header)
 
         if verify or not closed:
-            f = fitsopen(filename, mode='append')
+            f = fitsopen(filename, mode='append', **kwargs)
             try:
                 f.append(hdu)
-
                 # Set a flag in the HDU so that only this HDU gets a checksum
                 # when writing the file.
                 hdu._output_checksum = checksum
             finally:
                 f.close(closed=closed)
         else:
             f = _File(filename, mode='append')
-            try:
-                hdu._output_checksum = checksum
-                hdu._writeto(f)
-            finally:
-                f.close()
+            if kwargs:
+                raise TypeError('Unexpected **kwargs: %r' % kwargs)
```

Comment:
A warning is enough here, no need to raise en exception. See other warnings in the file (with `AstropyUserWarning`).

Reply:
Yup, issuing a warning instead of raising an error.